internal/clients/team7/frameworks/OpinionFramework.go

```diff
@@ -1,31 +1,61 @@
 package frameworks
 
-/*
-Base interface for all opinion frameworks. This can be used to define
-different behaviours of opinion formation giving the agent different behaviours.
-*/
-type IOpinionFramework[I, SN, O any] interface {
-	GetOpinion(inputs I, socialNetwork SN) O
-}
+import (
+	"math"
 
-type OpinionFrameworkInputs struct {
-}
+	"github.com/google/uuid"
+)
 
-type OpinionFrameworkOutputs struct {
+type OpinionFrameworkInputs struct {
+	AgentOpinion map[uuid.UUID]float64
+	Mindset      float64
 }
 
 type OpinionFramework struct {
-	IOpinionFramework[OpinionFrameworkInputs, SocialNetwork, OpinionFrameworkOutputs]
-	inputs *OpinionFrameworkInputs
+	Inputs *OpinionFrameworkInputs
 }
 
-// Constructor for OpinionFramework
 func NewOpinionFramework(of OpinionFrameworkInputs) *OpinionFramework {
-	return &OpinionFramework{inputs: &of}
+	return &OpinionFramework{Inputs: &of}
 }
 
-func (of *OpinionFramework) GetOpinion(inputs OpinionFrameworkInputs, socialNetwork SocialNetwork) OpinionFrameworkOutputs {
-	of.inputs = &inputs
-	// TODO: Formulate opinion
-	return OpinionFrameworkOutputs{}
+func (of *OpinionFramework) GetOpinion() float64 {
+
+	i := len(of.Inputs.AgentOpinion)
+	μ := of.Inputs.Mindset
+
+	O := make([]float64, i)
+	for idx := range O {
+		O[idx] = of.Inputs.AgentOpinion[uuid.UUID]
+	}
+
+	W := make([]float64, i)
+	for idx := range W {
+		W[idx] = 1
+	}
```

Comment:
This currently sets the weights to 1 each time while the algorithm is supposed to use the previous weights right? Probably won't get this done in time so we can leave it like this I suppose, but ideally we should store a weight for each different opinion right?